pkg/csi/client.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/kubernetes-csi/csi-lib-utils/protosanitizer"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+	"k8s.io/klog"
+)
+
+// Client is a gRPC client connect to remote CSI driver and abstracts all CSI calls.
+type Client interface {
+	// GetDriverName returns driver name as discovered by GetPluginInfo()
+	// gRPC call.
+	GetDriverName(ctx context.Context) (string, error)
+
+	// SupportsPluginControllerService return true if the CSI driver reports
+	// CONTROLLER_SERVICE in GetPluginCapabilities() gRPC call.
+	SupportsPluginControllerService(ctx context.Context) (bool, error)
+
+	// SupportsControllerResize returns whether the CSI driver reports EXPAND_VOLUME
+	// in ControllerGetCapabilities() gRPC call.
+	SupportsControllerResize(ctx context.Context) (bool, error)
+
+	// Expand expands the volume to a new size at least as big as requestBytes.
+	// It returns the new size and whether the volume need expand operation on the node.
+	Expand(ctx context.Context, volumeID string, requestBytes int64, secrets map[string]string) (int64, bool, error)
+
+	// Probe checks that the CSI driver is ready to process requests
+	Probe(ctx context.Context) error
+}
+
+// New creates a new CSI client.
+func New(address string, timeout time.Duration) (Client, error) {
+	conn, err := newGRPCConnection(address, timeout)
+	if err != nil {
```

Comment:
Lets use connection interface from https://github.com/kubernetes-csi/csi-lib-utils/pull/11 ?

Reply:
OK, these util functions make sense to me, I will update this later.